### PR TITLE
blocks docs

### PR DESCRIPTION
**Pull request by novaugust**
- one-lining simple rewrites
- accidentally wrote a manifesto instead of documenting block rewrites

### CODE REVIEW

File: docs/mix_configs.cheatmd [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## Mix Configs
+
+Mix Config files have their config stanzas sorted. Similar to the sorting of aliases, this delivers consistency to an otherwise arbitrary world, and can even help catch bugs like configuring the same key multiple times.
+
+A file is considered a config file if
+
+1. its path matches `config/.*\.exs` or `rel/overlays/.*\.exs`
+2. the file imports Mix.Config (`import Mix.Config`)
```
Comment (milmazz) [inline docs/mix_configs.cheatmd:8]:
```suggestion
2. the file imports Config (`import Config` or the deprecated `Mix.Config`)
```

File: docs/mix_configs.cheatmd [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+Once a file is detected as a mix config, its `config/2,3` stanzas are grouped and ordered like so:
+
+- group config stanzas separated by assignments (`x = y`) together
+- sort each group according to erlang term sorting
+- move all existing assignments between the config stanzas to above the stanzas (without changing their ordering)
```
Comment (milmazz) [inline docs/mix_configs.cheatmd:14]: Can you provide some examples? I thin that will make clear what are you talking about.

File: docs/module_directives.cheatmd [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+You're likely here just to see what Styler does, in which case, please [click here to skip](skip) following manifesto on our philosophy regarding the usage of these macros.
+
+## Which Control Flow Macro Should I Use?
+
+The number of "blocks" in Elixir means there are a large number of ways to write semantically equivalent code, often leaving developers [in the dark as to which structure they should use.](https://www.reddit.com/r/elixir/comments/1ctbtcl/i_am_completely_lost_when_it_comes_to_which/)
+
+We believe readability is enhanced by using the simplest api possible, whether we're talking about internal module function calls or standard-library macros.
+
+### `case`, `if`, & `unless`
```
[the remark this answers is not in the document]
Comment (novaugust) [reply docs/module_directives.cheatmd:14]: bahaha ty

File: docs/mix_configs.cheatmd (hunk 2 above)
Comment (novaugust) [reply docs/mix_configs.cheatmd:14]: yes sorry, this is WIP! examples are coming, rough notes atm :) just the manifesto re: blocks felt ready for review, sorry for not being clearer when i linked ya

File: docs/mix_configs.cheatmd (hunk 1 above)
Comment (novaugust) [reply docs/mix_configs.cheatmd:8]: ha actually it doesn't run on `import Mix.Config` xD

File: docs/module_directives.cheatmd [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+We advocate for `case` and `if` as the first tools to be considered for any control flow as they are the two simplest blocks. If a branch _can_ be expressed with an `if` statement, it _should_ be. Otherwise, `case` is the next best choice. In situations where developers might reach for an `if/elseif/else` block in other languages, `cond do` should be used.
+
+(`cond do` seems to see a paucity of use in the language, but many complex nested expressions or with statements can be improved by replacing them with a `cond do`).
+
+`unless` is a special case of `if` meant to make code read as natural-language (citation needed). While it sometimes succeeds in this goal, its absence in most programming languages often makes it feel cumbersome to programmers with non-Ruby backgrounds. Thankfully, with Styler's help developers don't need to ever reach for `unless` - expressions that are "simpler" with its use are automatically rewritten to use it.
+
```
[the remark this answers is not in the document]
Comment (novaugust) [reply docs/module_directives.cheatmd:21]: yeah thanks. need to figure out what / how much i want to communicate here. had meant to keep it _very_ high level but once i uncorked it kept flowing.

File: docs/control-flow_macros.cheatmd [hunk 5]
```diff
@@ -0,0 +1,47 @@
+# Control Flow Macros (aka "Blocks": `case`, `if`, `unless`, `cond`, `with`)
+
+Elixir's Kernel documentation refers to these structures as "macros for control-flow".
+We often refer to them as "blocks" in our changelog, which is a much worse name, to be sure.
+
+You're likely here just to see what Styler does, in which case, please [click here to skip](skip) following manifesto on our philosophy regarding the usage of these macros.
```
Comment (lancehalvorsen) [inline docs/control-flow_macros.cheatmd:6]: I think you need a "the" here -> the following manifesto

File: docs/control-flow_macros.cheatmd [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+## Which Control Flow Macro Should I Use?
+
+The number of "blocks" in Elixir means there are a large number of ways to write semantically equivalent code, often leaving developers [in the dark as to which structure they should use.](https://www.reddit.com/r/elixir/comments/1ctbtcl/i_am_completely_lost_when_it_comes_to_which/)
```
Comment (lancehalvorsen) [inline docs/control-flow_macros.cheatmd:10]: Consider simplifying this to remove the singular/plural subject verb agreement dissonance with "large number." Something like "means there are many ways to write"

File: docs/control-flow_macros.cheatmd [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+We believe readability is enhanced by using the simplest api possible, whether we're talking about internal module function calls or standard-library macros.
+
+### `case`, `if`, and `unless`
+
+We advocate for `case` and `if` as the first tools to be considered for any control flow as they are the two simplest blocks. If a branch _can_ be expressed with an `if` statement, it _should_ be. Otherwise, `case` is the next best choice. In situations where developers might reach for an `if/elseif/else` block in other languages, `cond do` should be used.
+
+(`cond do` seems to see a paucity of use in the language, but many complex nested expressions or with statements can be improved by replacing them with a `cond do`).
+
+`unless` is a special case of `if` meant to make code read as natural-language (citation needed). While it sometimes succeeds in this goal, its absence in most programming languages often makes it feel cumbersome to programmers with non-Ruby backgrounds. Thankfully, with Styler's help developers don't need to ever reach for `unless` - expressions that are "simpler" with its use are automatically rewritten to use it.
```
Comment (lancehalvorsen) [inline docs/control-flow_macros.cheatmd:20]: Add a comma here, I think: "Thankfully, with Styler's help, developers . . ."

Comment (novaugust) [reply docs/control-flow_macros.cheatmd:20]: augh tough one. i could go either way hah.

File: docs/control-flow_macros.cheatmd [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+### `with`
+
+> `with` great power comes great responsibility
+>
+> - Uncle Ben
+
+As the most powerful of the Kernel control-flow expressions, `with` requires the most cognitive overhead from readers to be understood. Its power means that any expression that can be expressed as a `case`/`if`/`cond` can _also_ be expressed as a `with` statement (especially with the liberal application of small private helper functions).
```
Comment (lancehalvorsen) [inline docs/control-flow_macros.cheatmd:28]: Maybe remove passive voice here? 

As the most powerful of the Kernel control-flow expressions, `with` requires the most cognitive overhead to understand. Its power means that we can use it as a replacement for anything we might express using a `case`, `if`, or `cond`(especially with the liberal application of small private helper functions).

File: docs/control-flow_macros.cheatmd (hunk 7 above)
Comment (novaugust) [reply docs/control-flow_macros.cheatmd:20]: i tried it w/ the comma and it didn't work for me. gonna leave as-is

File: docs/control-flow_macros.cheatmd [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+Unfortunately, this has lead to a proliferation of `with` in codebases where simpler expressions would have sufficed, meaning a lot of Elixir code ends up being harder for readers to understand than it needs to be.
+
+Thus, `with` is the control-flow structure of last resort. We advocate that `with` **should only be used when more basic expressions do not suffice or become overly verbose**. To qualify "overly verbose", we subscribe to the [Chris Keathley school of thought](https://www.youtube.com/watch?v=l-8ghbdRB1w) that judicious nesting of control flow blocks within a function isn't evil and more-often-than-not is superior to spreading implementation over many small single-use functions. We'd even go so far as to suggest that cyclomatic complexity is an inexact measure of code quality, with more than a few false negatives and many false positives.
```
Comment (lancehalvorsen) [inline docs/control-flow_macros.cheatmd:32]: Add an "as" here?: To qualify as "overly verbose"

File: docs/control-flow_macros.cheatmd (hunk 8 above)
Comment (novaugust) [reply docs/control-flow_macros.cheatmd]: lance there's no passive verb in there :p

File: docs/control-flow_macros.cheatmd (hunk 9 above)
Comment (novaugust) [reply docs/control-flow_macros.cheatmd]: ah i meant "to qualify what i mean by overly verbose..."

Comment (novaugust) [reply docs/control-flow_macros.cheatmd]: >As for verbosity, we subscribe to the

?

**Comment by novaugust** [comment]
thanks @milmazz and @lancehalvorsen for all your feedback! i'm going to merge as is since main's docs are in a WIP state anyways, i'd be delighted if you dm'd me or PR'd against what i've got so far =D

**Pull request merged**